src/hoi4_world/technology/technologies.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SRC_HOI4WORLD_TECHNOLOGY_TECHNOLOGIES_H_
+#define SRC_HOI4WORLD_TECHNOLOGY_TECHNOLOGIES_H_
+
+
+
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "src/mappers/technology/tech_mapping.h"
+
+
+
+namespace hoi4
+{
+
+class Technologies
+{
+  public:
+   Technologies() = default;
+
+   explicit Technologies(std::map<std::optional<std::string>, std::set<std::string>> technologies_by_limits):
+       technologies_by_limits_(std::move(technologies_by_limits))
+   {
+   }
+
+   [[nodiscard]] bool HasTechnology(const std::string& technology) const;
+
+   [[nodiscard]] const std::map<std::optional<std::string>, std::set<std::string>>& GetTechnologies() const
+   {
+      return technologies_by_limits_;
+   }
+
+   std::partial_ordering operator<=>(const Technologies&) const = default;
+
+  private:
+   // technologies are sorted by limits that can be applied to them
+   std::map<std::optional<std::string>, std::set<std::string>> technologies_by_limits_;
```

Comment:
You are using an optional as a key.. Are you certain this is what you want? Multiple nullopts won't be inserted separately.

Reply:
Are you sure? From en.cppreference.com operator == work like so:

> Compares two optional objects, lhs and rhs. The contained values are compared (using the corresponding operator of T) only if both lhs and rhs contain values. Otherwise,
**lhs is considered equal to rhs if, and only if, both lhs and rhs do not contain a value.**
lhs is considered less than rhs if, and only if, rhs contains a value and lhs does not.

That implies nullopts should be equal.

Reply:
Wrote a quick test to double-check and they are.